packages/ai-openai/src/node/openai-language-model.ts

```diff
@@ -131,32 +139,35 @@ export class OpenAiModel implements LanguageModel {
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             resolve?.(runner.finalChatCompletion as any);
         });
+        runner.on('chunk', chunk => {
+            if (cancellationToken?.isCancellationRequested) {
+                resolve = undefined;
+                reject = undefined;
```

Comment:
See my comments re: the current pinned resolver approach in #14920. Let's decide whether it could really cause the problems I think it could before we move forward.